DeepLinkKit/Router/DPLDeepLinkRouter.m

```diff
@@ -147,45 +143,84 @@ - (BOOL)handleUserActivity:(NSUserActivity *)userActivity withCompletion:(DPLRou
     return NO;
 }
 
+- (UIViewController <DPLTargetViewController> *)viewControllerForUrl:(NSURL *)url {
+    DPLDeepLink *deepLink;
+    id handler;
+    deepLink = [self deepLinkForUrl:url handler:&handler];
+    if (handler) {
+        DPLRouteHandler *routeHandler = [self routeHandlerForHandler:handler];
+        if (routeHandler) {
+            return [self viewControllerForHandler:routeHandler withDeepLink:deepLink];
+        }
+    }
+    return nil;
+}
 
-- (BOOL)handleRoute:(NSString *)route withDeepLink:(DPLDeepLink *)deepLink error:(NSError *__autoreleasing *)error {
-    id handler = self[route];
-    
-    if ([handler isKindOfClass:NSClassFromString(@"NSBlock")]) {
-        DPLRouteHandlerBlock routeHandlerBlock = handler;
-        routeHandlerBlock(deepLink);
+
+- (DPLDeepLink *)deepLinkForUrl:(NSURL *)url handler:(id *)handler {
```

Comment:
I agree that an inout parameter isn't ideal here. Maybe this method could return a dictionary? Or we could even create a lightweight class that had `deepLink` and `handler` properties.